[Fire Mage] Combustion Updates

Sharrq opened the pull request:
Updated Combustion Spell Usage to the new event listeners and added support for Blaster Master

Dambroda commented on src/parser/mage/fire/modules/features/CombustionSpellUsage.js:
```diff
@@ -47,7 +69,7 @@ class CombustionSpellUsage extends Analyzer {
     when(this.suggestionThresholds)
       .addSuggestion((suggest, actual, recommended) => {
-        return suggest(<>You cast <SpellLink id={SPELLS.FIREBALL.id} /> or <SpellLink id={SPELLS.SCORCH.id} /> {this.castedWithInstants} times ({this.castsWithInstantsPerCombustion.toFixed(2)} per Combustion) while you had charges of <SpellLink id={SPELLS.FIRE_BLAST.id} /> or <SpellLink id={SPELLS.PHOENIX_FLAMES_TALENT.id} /> available. Make sure you are using up all of your charges of Fire Blast ...
+        return suggest(<>You started to cast <SpellLink id={SPELLS.FIREBALL.id} /> {!this.hasBlasterMaster ? <> or <SpellLink id={SPELLS.SCORCH.id} /></> : '' } {this.badCastsStarted} times ({this.castsWithInstantsPerCombustion.toFixed(2)} per Combustion), and completed {this.badCastsCompleted} casts, while you had charges of <SpellLink id={SPELLS.FIRE_BLAST.id} /> {this.hasPhoenixFlames ? <> or  ...
```
Some unneeded fragments (<>) in here, also maybe it would be neater to have it spread out on multiple lines ?

Sharrq replied:
The spell link inside the ternary won't work without the extra fragment. That's why it's there

Dambroda commented on src/parser/mage/fire/modules/features/CombustionSpellUsage.js:
```diff
@@ -14,22 +15,43 @@ class CombustionSpellUsage extends Analyzer {
-      this.castedWithInstants += 1;
-      debug && this.log("Casted with Instants Available");
+
+    if (event.type === "cast" && (fireBlastCharges > 0 || phoenixFlamesCharges > 0)) {
```
You have two instances of ` && (fireBlastCharges > 0 || phoenixFlamesCharges > 0)`

Instead should probably (before the rest), just do `if(fireBlastCharges < 1  && phoenixFlamesCharges < 1) { return; }`

Sharrq commented:
I made the change for the blast/phoenix charges, the extra fragments was intentional to make the text/spell links in the ternary work.

Dambroda commented:
Looks good, and I saw you talked to the TCS about how best to represent this suggestion. So I'll wait until I hear from you about that

Sharrq commented:
Yea im trying to think it through and figure out how i want to do it, cause im afraid it will over complicate things if i also need to check at the end of combustion, but disregard PF charges gained within x seconds of combustion ending, and ignore fire blast procs gained within like 500ms of combustion ending, and if they got lucky with pyroclasm ... so on and so forth .....

So im trying to decide what i can do without making it insanely complicated.

It might be worth removing that check entirely and just yell at them for casting Fireball during Combustion.... But what they do inside combustion matters too ... so im not sure.

Sharrq commented:
I adjusted this to count Fireball and Scorch separately and to count any fireball cast during Combustion as a mistake. Also added a separate checklist item and suggestion for Fireball.

Sharrq commented:
this should be ready to go now.

Dambroda commented on src/parser/mage/fire/modules/Checklist/Component.js:
```diff
@@ -59,7 +59,8 @@ class FireMageChecklist extends React.PureComponent {
           {combatant.hasTalent(SPELLS.FIRESTARTER_TALENT.id) && <Requirement name="Pyroclasm Usage" thresholds={thresholds.pyroclasmCombustionUsage} tooltip="If you have enough time to complete the cast before Combustion ends, then you should always use your Pyroclasm procs to hard cast Pyroblast during Combustion. This is primarily because Combustion will guarantee that the spell crits, resultin ...
-          <Requirement name="Bad Spell Usage" thresholds={thresholds.combustionSpellUsage} tooltip="It is very important to use your time during Combustion wisely to get as many Hot Streak procs as possible before Combustion ends. To accomplish this, you should be stringing your guaranteed crit spells (Fireblast and Phoenix Flames) together to perpetually convert Heating Up to Hot Streak as many  ...
+          <Requirement name="Bad Scorch Uses" thresholds={thresholds.scorchSpellUsageDuringCombustion} tooltip="It is very important to use your time during Combustion wisely to get as many Hot Streak procs as possible before Combustion ends. To accomplish this, you should be stringing your guaranteed crit spells (Fireblast and Phoenix Flames) together to perpetually convert Heating Up to Hot Str ...
+          <Requirement name="Bad Fireball Uses" thresholds={thresholds.fireballSpellUsageDuringCombustion} tooltip="Due to Combustion's short duration, you should never cast Fireball during Combustion. Instead, you should use your instant cast abilities like Fireblast and Phoenix Flames. If you run out of instant abilities, cast Scorch instead since it's cast time is shorter." />
```
```suggestion
          <Requirement name="Bad Fireball Uses" thresholds={thresholds.fireballSpellUsageDuringCombustion} tooltip="Due to Combustion's short duration, you should never cast Fireball during it. Instead, you should use your instant cast abilities like Fireblast and Phoenix Flames. If you run out of instant abilities, cast Scorch instead because its cast time is shorter." />
```
